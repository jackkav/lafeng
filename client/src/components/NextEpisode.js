import React from 'react'
import styled from 'styled-components'

const NextEpisode = ({ children, onClick, release, name, number }) => (
  <Container onClick={onClick}>
    <Image />
    <TextContainer>
      <InnerTextContainer>
        <EpisodeNumber>{number}</EpisodeNumber>
        <ReleaseDay>{release}</ReleaseDay>
      </InnerTextContainer>
      <SeriesName>{children}</SeriesName>
      <EpisodeName>{name}</EpisodeName>
    </TextContainer>
  </Container>
)

NextEpisode.defaultProps = {
  children: 'Game of Thrones',
  releaseDate: 'Today',
  episodeNumber: 'S01E01',
  episodeName: 'Name'
}
NextEpisode.propTypes = {
  children: React.PropTypes.string.isRequired,
  name: React.PropTypes.string,
  number: React.PropTypes.string,
  onClick: React.PropTypes.func,
  release: React.PropTypes.string
}

const Container = styled.div`
  display: flex;
  flex: 1;
  flexDirection: row;
  height: 150px;
  justify-content: center;
  cursor: pointer;
  &:hover {
    color: grey;
  }
`
const Image = styled.div`
  flex: 1;
  background-color: powderblue;
`
const TextContainer = styled.div`
  display: flex;
  flex-direction: column;
  flex: 4;
  background-color: grey;
  font-family: Helvetica, Arial, sans-serif;
`
const InnerTextContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex: 1;
`
const EpisodeNumber = styled.div`
  display: flex;
  align-items: center;
  flex: 3;
  background-color: yellow;
`
const EpisodeName = styled.div`
  display: flex;
  align-items: center;
  flex: 1;
  background-color: lightgreen;
`
const ReleaseDay = styled.div`
  display: flex;
  align-items: center;
  flex: 1;
  background-color: orange;
  justify-content: center;
`
const SeriesName = styled.div`
  display: flex;
  align-items: center;
  flex: 1;
  background-color: darkgreen;
`

export default NextEpisode
