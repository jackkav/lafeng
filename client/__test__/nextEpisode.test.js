import React from 'react'
import NextEpisode from '../src/components/NextEpisode'
import renderer from 'react-test-renderer'

it('renders correctly without any props', () => {
  const tree = renderer.create(
    <NextEpisode />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
