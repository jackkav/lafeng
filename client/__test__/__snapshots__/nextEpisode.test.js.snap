exports[`test renders correctly without any props 1`] = `
<div
  className="gZhfdt"
  onClick={undefined}>
  <div
    className="gxzOve" />
  <div
    className="fRItGO">
    <div
      className="cVrTOJ">
      <div
        className="bDZGqW" />
      <div
        className="cGprWl" />
    </div>
    <div
      className="gTnksx">
      Game of Thrones
    </div>
    <div
      className="gCMJK" />
  </div>
</div>
`;
